### DISCUSSION TITLE
Youtube Bug

### DISCUSSION THREAD

**Discussion by FarhadiAlireza** [GFW-knocker/gfw_resist_tls_proxy#77] [General]
_Hi. I found the below log about the youtube bug
I guess it may help you, so I share it_
-Love U guys-

**Audit usage of navigator.userAgent, navigator.appVersion, and navigator.platform

A page or script is accessing at least one of navigator.userAgent, navigator.appVersion, and navigator.platform. Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.
To fix this issue, replace the usage of navigator.userAgent, navigator.appVersion, and navigator.platform with feature detection, progressive enhancement, or migrate to navigator.userAgentData.
Note that for performance reasons, only the first access to one of the properties is shown.
1 source
content.js:1
User-Agent String Reduction

-------------------------------------------------------------------

Ensure CORS response header values are valid

A cross-origin resource sharing (CORS) request was blocked because of invalid or missing response headers of the request or the associated preflight request .
To fix this issue, ensure the response to the CORS request and/or the associated preflight request are not missing headers and use valid header values.
Note that if an opaque response is sufficient, the request's mode can be set to no-cors to fetch the resource with CORS disabled; that way CORS headers are not required but the response content is inaccessible (opaque).
2 requests
Request	Status	Preflight Request (if problematic)	Header	Problem	Invalid Value (if available)
 Create	blocked	
 Create	Access-Control-Allow-Origin	Missing Header	
 Create	blocked	
 Create	Access-Control-Allow-Origin	Missing Header	

Cross-Origin Resource Sharing (CORS)**

**Comment by GFW-knocker** [comment]
@FarhadiAlireza 
what brilliant guys we have
i learn from you & i love you too.

**Reply by FarhadiAlireza** [reply]
❤️❤️❤️

**Reply by GFW-knocker** [reply]
can you tell me by which software you do audit?

**Reply by FarhadiAlireza** [reply]
I ran the Python script and inserted the JSON config in V2ray
then open Chorom, Reach Youtube.com, and watch a clip, click F12
it's better to see the below picture

![Screenshot 2023-04-30 004923](https://user-images.githubusercontent.com/39120022/235324751-d8f0de8e-2851-4645-beb0-50c13e500147.jpg)

**Reply by GFW-knocker** [reply]
i didnt aware of such options.
thanks again.

**Reply by FarhadiAlireza** [reply]
Is it usefull ? 🤔

**Reply by GFW-knocker** [reply]
sure.
now i am audit every page and quickly find root of problem
thank you
